plugins/catalog-import/src/components/EntityListComponent/EntityListComponent.tsx

```diff
@@ -0,0 +1,163 @@
+import PersonIcon from '@material-ui/icons/Person';
+import WorkIcon from '@material-ui/icons/Work';
+import React, { useState } from 'react';
+
+const useStyles = makeStyles(theme => ({
+  nested: {
+    paddingLeft: theme.spacing(4),
+  },
+}));
+
+function sortEntities(entities: Array<EntityName | Entity>) {
+  return entities.sort((a, b) =>
+    formatEntityRefTitle(a).localeCompare(formatEntityRefTitle(b)),
+  );
+}
+
+function getEntityIcon(entity: { kind: string }): React.ReactElement {
+  switch (entity.kind.toLowerCase()) {
+    case 'api':
+      return <ExtensionIcon />;
+
+    case 'component':
+      return <MemoryIcon />;
+
+    case 'domain':
+      return <ApartmentIcon />;
+
+    case 'location':
+      return <LocationOnIcon />;
+
+    case 'group':
+      return <GroupIcon />;
+
+    case 'system':
+      return <CategoryIcon />;
+
+    case 'user':
+      return <PersonIcon />;
+
+    default:
+      return <WorkIcon />;
+  }
+}
+
+export const EntityListComponent = ({
+  locations,
+  collapsed = false,
+  locationListItemIcon,
+  onItemClick,
+  firstListItem,
+  withLinks = false,
+}: {
+  locations: Array<{ target: string; entities: (Entity | EntityName)[] }>;
+  locationListItemIcon: (target: string) => React.ReactElement;
+  collapsed?: boolean;
+  firstListItem?: React.ReactElement;
+  onItemClick?: (target: string) => void;
+  withLinks?: boolean;
+}) => {
+  const classes = useStyles();
+
+  const [expandedUrls, setExpandedUrls] = useState<string[]>([]);
+
+  const handleClick = (url: string) => {
+    setExpandedUrls(urls =>
+      urls.includes(url) ? urls.filter(u => u !== url) : urls.concat(url),
+    );
+  };
+
+  return (
+    <List>
+      {firstListItem}
+      {locations.map(r => (
+        <React.Fragment key={r.target}>
+          <ListItem
+            dense
+            button={Boolean(onItemClick) as any}
+            onClick={() => onItemClick?.call(this, r.target)}
+          >
+            <ListItemIcon>{locationListItemIcon(r.target)}</ListItemIcon>
+
+            <ListItemText
+              primary={r.target}
+              secondary={`Entities: ${r.entities.length}`}
+            />
+
+            {collapsed && (
+              <ListItemSecondaryAction>
+                <IconButton edge="end" onClick={() => handleClick(r.target)}>
+                  {expandedUrls.includes(r.target) ? (
+                    <ExpandLessIcon />
+                  ) : (
+                    <ExpandMoreIcon />
+                  )}
+                </IconButton>
+              </ListItemSecondaryAction>
+            )}
+          </ListItem>
+
+          <Collapse
+            in={!collapsed || expandedUrls.includes(r.target)}
+            timeout="auto"
+            unmountOnExit
+          >
+            <List component="div" disablePadding dense>
+              {sortEntities(r.entities).map(entity => (
+                <ListItem
+                  component={withLinks ? EntityRefLink : 'li'}
```

Comment:
It also replaces the `li` with a link in the `withLinks` case which is also suboptimal. I can change it to always use a div since the semantic is also broken on another level, since we have this: `<ul><li>Parent</li><ul><li>Child</li></ul></ul>` (The `Collapse` makes it worse by doing: `<ul><li>Parent</li><div><ul><li>Child</li></ul></div></ul>`), while this would be correct: `<ul><li>Parent<ul><li>Child</li></ul></li></ul>`.

Since `<a …><li>…</li></a>` is also not semantically correct, I wonder how to best make the link clickable, without doing it manually with `onClick()` (though the `button` property also switches the `li` to a `div`…).